### PR TITLE
Bug list size overload

### LINKED ISSUE #29: Add two new endpoints to E2E
Add presigned upload URL and download script endpoints to E2E testing postman collection.

### LINKED ISSUE #35: Remove Postgress env from e2e yml file
Remove Postgress env from the e2e yml file until FGAC is incorporated in the e2e testing.

### LINKED ISSUE #36: Optimize `GetList` Function for Large Dataset Handling
The current implementation of `GetList` in our `S3Controller` retrieves and accumulates all objects corresponding to a specific bucket and prefix. This approach is not memory-efficient for large datasets, leading to potential server overload. The goal of this enhancement is to optimize `GetList` to handle large datasets more efficiently by processing each page of results individually.

### PR DESCRIPTION
**Pull request by Akopti8**
- Enhance logging to match standards. Revolves #28 
- Introduced `GetListWithCallBack` for safer retrieval of object lists, reducing memory usage compared to `GetList`. Resolves #36 
- Remove unnecessary Postgres from e2e. Resolves #35 
- Add new endpoints to e2e testing. Resolves #29

### CODE REVIEW

File: blobstore/presigned_url.go [hunk 1]
```diff
@@ -232,111 +238,92 @@ func (bh *BlobHandler) HandleGetPresignedURLMultiObj(c echo.Context) error {
...
+	log.Info("successfully generated presigned URL for prefix:", prefix)
 	return c.JSON(http.StatusOK, string(href))
 }
 
 func (bh *BlobHandler) HandleGenerateDownloadScript(c echo.Context) error {
 	prefix := c.QueryParam("prefix")
-	bucket := c.QueryParam("bucket")
-	if prefix == "" || bucket == "" {
+	if prefix == "" {
 		errMsg := fmt.Errorf("`prefix` and `bucket` query params are required")
```
Comment (ShaneMPutnam) [inline blobstore/presigned_url.go:260]: Update error message, since you are only checking the prefix here?

File: blobstore/presigned_url.go [hunk 2]
```diff
@@ -232,111 +238,92 @@ func (bh *BlobHandler) HandleGetPresignedURLMultiObj(c echo.Context) error {
...
-	}
-
-	//expiration period from the env
-
+	var totalSize uint64
 	var scriptBuilder strings.Builder
 	createdDirs := make(map[string]bool)
-	// Add download instructions at the beginning of the script
+	basePrefix := filepath.Base(strings.TrimSuffix(prefix, "/"))
+	scriptBuilder.WriteString(fmt.Sprintf("mkdir \"%s\"\n", basePrefix))
```
Comment (ShaneMPutnam) [inline blobstore/presigned_url.go:276]: I think line 276 was copied by accident?

**Review by ShaneMPutnam — CHANGES_REQUESTED** [review]
Looks of nice updates in here! See inline comments and some general comments here:

- There are a few places where you use errors.New instead of fmt.Errorf. See line 145 of move.go for example. Try to be consistent
- Sometimes your errors/return messages start with lower case letter and other times they start with a capital letter.
- Most of the time you use a colon followed by the error message, though sometimes you use a comma. 

Note that I skimmed the test scripts, but did not test them since I am not sure how to do that. Also I do not totally understand the logic behind how the test names are numbered, at times they seem to be inconsistent, but I could just not get it.

File: blobstore/move.go [hunk 3]
```diff
@@ -47,58 +48,84 @@ func (bh *BlobHandler) HandleMovePrefix(c echo.Context) error {
...
-	}
+	var objectsFound bool
+
+	processPage := func(page *s3.ListObjectsV2Output) error {
+		if len(page.Contents) == 0 {
+			return nil // No objects to process in this page
+		}
+		objectsFound = true // Objects found, set the flag
+
+		for _, object := range page.Contents {
```
[the remark this answers is not in the document]
Comment (Akopti8) [reply blobstore/move.go:59]: Last time I was working on this endpoint, I asked whether we wanted to have these handle multi-buckets and I was told to wait on that. 
In the future when we have extra time, I am planning on exposing this endpoint in the bucketview UI and I can revisit concurrency and multibucket capabilities. If you're okay with it, I will make this into an issue.

File: blobstore/list.go [hunk 4]
```diff
@@ -27,6 +27,31 @@ type ListResult struct {
 	ModifiedBy string    `json:"modified_by"`
 }
 
+// CheckAndAdjustPrefix checks if the prefix is an object and adjusts the prefix accordingly.
+// Returns the adjusted prefix, an error message (if any), and the HTTP status code.
+func CheckAndAdjustPrefix(s3Ctrl *S3Controller, bucket, prefix string) (string, string, int) {
```
Comment (ShaneMPutnam) [inline blobstore/list.go:32]: I think it would be more standard to return an error, instead of a errorMessage? Like (string, int, error)

**Review by ShaneMPutnam — APPROVED** [review]
Thanks for those updates! I made a few small changes: 0c976c9527c50421a7185f6a49ccf9d97d1401ef and one inline comment to consider as a part of your next PR. I also added or modified the following two issues for future updates:
- https://github.com/Dewberry/s3api/issues/44
- https://github.com/Dewberry/s3api/issues/45

**Pull request merged**